### PR TITLE
Phase to jobs schema change

### PR DESCRIPTION
**Pull request by chcosta**
Fixes https://github.com/dotnet/core-eng/issues/4687

FYI @riarenas

### CODE REVIEW

File: Documentation/AzureDevOps/PhaseToJobSchemaChange.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+azure-pipelines.yml
+
+```yml
+trigger:
+- master
+pr:
+- master
+
+phases:
```
[the remark this answers is not in the document]
Comment (chcosta) [reply Documentation/AzureDevOps/PhaseToJobSchemaChange.md]: ack

**Comment by chcosta** [comment]
FYI @garath

**Comment by chcosta** [comment]
@markwilkie thank you for the zero value add!

**Comment by jcagme** [comment]
cc/ beats FYI all the way

**Comment by riarenas** [comment]
LGTM. Once we make the change in the samples repo we can add a link to its new yaml file in addition to the Arcade one.

**Comment by chcosta** [comment]
You mean like add more to https://github.com/dotnet/arcade/blob/c86dcfc3ac5bc1c030931cd7328385bfdefa9af7/Documentation/AzureDevOps/PhaseToJobSchemaChange.md#phase-to-job-schema-transition?  I called out the interesting ones but I assumed the template itself was commented enough that someone could figure out the rest

**Comment by ChadNedzlek** [comment]
Well, several of them just say "Include SuchAndSuch task"... but that doesn't mean anything to me who isn't already familiar with arcade.  I don't know what "micro build" is, or what the "PublishTestResults" task is or does.  Or why I'd care about the "_HelixSource" variable. Or what the "derived dependsOn value for "publish build assets"" means.

**Comment by chcosta** [comment]
Fair, but this document is *intended* to be targeted towards people transitioning from `phases\base.yml` to `jobs\jobs.yml` so those concepts should already be familiar.  The problem here is that I likely went down the path of over-documenting somethings here and that makes it feel more general / relevant than it actually is.  I'm happy to chat about these things in person though.

**Comment by garath** [comment]
I think the "over-documented" parts are best (eventually) moved to AzureDevOpsOnboarding.md to keep the scope of this document narrow. Expansion and linking Onboarding will address Chad's concerns.

**Comment by chcosta** [comment]
Merged so that @garath will stop closing / re-opening my PR

**Pull request merged**